### PR TITLE
Ben/docker and docker compose

### PR DESCRIPTION
**Pull request by benzcash**
- Added Dockerfile
- Added `docker-compose.yml` for developing with docker-compose
- Updated README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,11 +2,82 @@
...
 
 ```
 go get -v github.com/zcash-hackworks/zcashd_exporter
 go install github.com/zcash-hackworks/zcashd_exporter
 $GOPATH/bin/zcashd_exporter --help
 ```
 
+## Getting Started With docker-compose
+
+#### Create zcashd direcotries
```
Comment (charlieok) [inline README.md]: I was thrown a bit by the jump from H2 to H4 headers. Should this one (and the following one) be H3?

File: docker/prometheus/config.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
+rule_files:
+  # - "first_rules.yml"
+  # - "second_rules.yml"
+  
+# A scrape configuration containing exactly one endpoint to scrape:
+# Here it's Prometheus itself.
+scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: 'prometheus'
```
Comment (charlieok) [inline docker/prometheus/config.yml]: I'm curious what it means for prometheus to scrape itself

**Review by charlieok — APPROVED** [review]
ACK

File: README.md (hunk 1 above)
Comment (benzcash) [reply README.md]: Ah, fixed the headings

File: docker/prometheus/config.yml (hunk 2 above)
Comment (benzcash) [reply docker/prometheus/config.yml]: Just some self monitoring from the default prometheus config.

Removed!

**Pull request merged**